conans/model/layout.py

```diff
@@ -0,0 +1,90 @@
+import os
+
+
+class _LayoutEntry(object):
+
+    def __init__(self):
+        self.folder = ""
+        self.includedirs = {}
+        self.builddirs = {}
+        self.resdirs = {}
+        self.libdirs = {}
+        self.bindirs = {}
+
+
+class Layout(object):
+    def __init__(self):
+
+        self._base_source_folder = None
+        self._base_build_folder = None
+        self._base_install_folder = None
+        self._base_package_folder = None
+
+        self.install = _LayoutEntry()
+        self.install.folder = ""
+
+        self.build = _LayoutEntry()
+        self.build.folder = ""  # Where the software is built (relative to _base_build_folder)
+        self.build.libdirs[""] = ["*.so", "*.so.*", "*.a", "*.lib", "*.dylib"]
+        self.build.bindirs[""] = ["*.exe", "*.dll"]
+
+        self.source = _LayoutEntry()
+        self.source.folder = ""
+        self.source.includedirs[""] = ["*.h", "*.hpp"]
+
+        self.package = _LayoutEntry()  # Where the artifacts are installed
+        self.package.includedirs["include"] = ["*"]
+        self.package.bindirs["bin"] = ["*"]
+        self.package.libdirs["lib"] = ["*"]
+
+    def __str__(self):
```

Comment:
AS this seems apparently for debugging purposes, I would convert it to a ``__repr__`` and include the class name in it.